radis/spectrum/spectrum.py

```diff
@@ -2404,15 +2404,20 @@ def apply_slit(
             if norm_by == "area":
                 self.units[q] = self.units[qns]
             elif norm_by == "max":
-                new_unit = "{0}*{1}".format(
-                    self.units[qns], unit.replace("cm-1", "cm-1")
-                )
+                if self.units[qns] == "1":
+                    new_unit = "{0}*{1}".format(
+                        self.units[qns], unit.replace("cm-1", "cm-1")
+                    )
+                else:
+                    new_unit = "{0}*{1}".format(
+                        unit.replace("cm-1", "cm-1"), self.units[qns]
+                    )
```

Comment:
Comments on the code:

- you should look up these `replace("cm-1", "cm-1")`  (they come from "cm_1" > "cm-1" ) and remove them. 

- do we really need the if/else here? 

---

There is also an underlying physical question, which requires to understand how experiments are done [ this will become a different Issue ]  : 

In my [comment](https://github.com/radis/radis/pull/106#issuecomment-615469626) I explained why the unit changes when you apply an instrumental slit function (to account for the spectrometer limited spectral resolution), if this slit function was normalized at the maximum instead of per area [ note that 99% of the people normalize at the area, we're dealing with a very particular case here, used only when calculating emission spectra to have an idea of the spectral flux]. This applied well to emission spectra.

Here, units == "1" refers to a transmittance spectra (or emissivity, but a similar reasoning applies): 

A theoretical transmittance spectrum is the ratio of an intensity I and a background I0. In practice, no-one ever measures a transmission spectrum directly. Your instrument collects an intensity I first, a background intensity I0 next, divides both and returns a transmittance spectrum I/I0. So your spectral resolution loss applies to I and I0. Your real transmittance spectrum is : `Filter(I) / Filter(I0)` instead of `Filter(I/I0)` as we are doing here. 
In practice the background is quite flat, so we have `Filter(I0)` ~ `I0` , so we end up with `Filter(I)/Filter(I0) ~ Filter(I)/I0 = Filter(I/I0)` , so applying the spectral filter to the theoretical transmittance spectrum directly is eventually a very good approximation.  But notice that  `Filter(I0)` ~ `I0`  assumed the filtering did not change the units! 

Therefore: 

- applying a slit not normalized by area to a transmittance spectrum is meaningless.  (people experienced in absorption spectroscopy like @dcmvdbekerom  will probably confirm!). 

- we want to remain numerically correct no matter what, so we should keep on correcting the unit as implemented here.  

- but we'll end up with transmittance spectra in "nm" or "cm-1" and with values that are not within [0 - 1] anymore: good luck for the user to understand that. Eventually, I think it would be appropriate to refactor apply_slit() and raise an error if we're applying a slit not normalized by area to a Spectrum that contains ratios (like Transmittance, but also Emissivity).  [ note that this is a different Issue ! ] 

What do you think?

Reply:
Hi @erwanp , thanks for the detailed comment. That's a lot of new information to process so I'll take it up one at a time!
Regarding why the if-else is necessary, that's a thing with astropy where concatenating strings works as follows:
```
u.Unit("1*cm")    #works fine   --> parses as u.cm
u.Unit("cm*1")    #throws error

u.Unit("mW/sr*cm")    #parses it as mW / (sr cm)
u.Unit("cm*mW/sr")    #parses it as (cm mW) / sr   --> what we want in the code
```

Regarding your second point, while I did understand the motivation and procedure behind how the experiment is conducted, I could not see how that connects with the fact that applying slit without area normalization is pointless. I did some research on the internet and the most commonly brought up argument is that peak/max normalization leads to erroneous results? Is that true?
I was also wondering if while calculating the transmittance spectra, could we perhaps allow the user to pass the intensity and background intensity separately with units? Wouldn't this help us drop "I/I0" as placeholders completely?
> but we'll end up with transmittance spectra in "nm" or "cm-1" and with values that are not within [0 - 1] anymore

Can't we normalize the spectrum after passing it through the slit?

Also apologies for any naive questions! While I did take up a few physics courses in my freshman and sophomore year, they were clearly not enough!

Reply:
Regarding your first point `u.Unit("cm*1")`: it's a nice trick, but looks like a Hack! What about replacing "1" with the empty string "" ?  It seems that that's how Astropy writes u.dimensionless_unscaled units anyway: 
```
u.Unit("1").__str__()
>> ""
u.Unit("").__str__()
>> ""
u.Unit("") == u.dimensionless_unscaled
>> True
```
And no need for the Hack then! 

---

Second point:

> the most commonly brought up argument is that peak/max normalization leads to erroneous results

It's not erroneous for emission emission spectra (if you account for the change in units), and sometimes useful: if what you're interested is the maximum energy Flux, for instance in radiative transfer, you'd get a result directly in energy flux  "~W/cm2/sr", instead of spectral energy flux "~W/cm2/sr/nm" ). 
For absorption spectrum, it's erroenous as I described it in my previous comment: because the underlying quantity is a ratio, so we assume that the units of the background spectrum did not change. 

What you're describing (given intensities separately) is possible. Actually, users could do :
```
s1 = Radiance(emission_spectrum)
s2 = Radiance(background_spectrum)
transmittance = s1/s2               # I'm not sure if this work for Spectrum objects, but same is possible using s1.get(), s2.get() ? 
```
But usually your instrument, or the instrument software, will process this directly and return a transmittance spectrum directly. So the main use case takes "I/I0". 

---

>  Can't we normalize the spectrum after passing it through the slit?

We could but that's not what the user asked for, and it would become very dangerous! We better raise an error saying that applying a non-area-normalized slit to a transmittance spectrum is erroneous, and that he should do otherwise. 

We could suggest a solution in the error message: 

```
# during error handling: 
if 'radiance_noslit' in self.get_vars():
    raise ValueError("ValueError("""applying a non-area-normalized slit to a transmittance spectrum is erroneous. Use `method="area"` in apply_slit(). If you're only interested in the emission spectrum, you can extract the emission spectrum the first: `s_rad = Radiance_noslit(s).apply_slit(...)`""")
else:
    raise ValueError("ValueError("""applying a non-area-normalized slit to a transmittance spectrum is erroneous. Use `method="area"` in apply_slit()""")

```

---

Hope this is clear! 
Also whatever physics class you took does not really matter here: it's more "technicalities" (for instance, knowledge of instruments & instrument software) than physics and science at this point! :)

Reply:
Hi Erwan, thanks for the detailed reply! I understand it much better now. 
Regarding the "hack", while we could use "" instead of "1" while creating `new_unit`, but we'd still be forced to use an if-else block (albeit less complex than the current one) since even after replacing "1" with "" will result in a string such as `cm*` which when parsed by `u.Unit` returns an error. I have made this change. Let me know if you think we could do this in a better way!

For the second point, I think I can see more clearly why you'd want the user to not call the `slit` method when using `max` normalization. While I wouldn't say it's all perfectly clear, the idea and the intuition behind the problem (and it's solution) have become a lot more solid now.
Finally, the code you shared (to give an error in case the user tries to call the `apply_slit` method incorrectly) looks good to me. I think we should do that once we're done with this.

Reply:
You're right about the `*`! 
Then what about this? :
```
new_unit = (u.Unit(self.units[qns]) * u.Unit(unit)).to_string()
```

Adimensioned.to_string() returns `""` so we should replace `"1"` with `""` for consistency